browser-sign-in/OktaBrowserSignIn/SingInViewController.swift

```diff
@@ -145,5 +141,63 @@ private extension SingInViewController {
         controller.title = title
         self.navigationController?.pushViewController(controller, animated: true)
     }
+    
+    func handleOktaAuthSuccess() {
+        self.hideProgress()
+        self.showAlert(title: "Signed In!")
+        self.updateUI()
+    }
+    
+    func handleOktaAuthLogoutSuccess() {
+        self.authState?.clear()
+        self.hideProgress()
+        self.showAlert(title: "Signed Out!")
+        self.updateUI()
+    }
+    
+    func handleOktaAuthFailure(error: Error) {
+        self.hideProgress()
+        self.showError(message: error.localizedDescription)
+    }
+}
+
+// UI Tests
+private extension SingInViewController {
+    
+    func signInTappedForUITests() -> Bool {
+        guard let config = configForUITests else { return true }
+        OktaAuth.signInWithBrowser().start(withDictConfig: config, view: self).then { _ in
+            self.handleOktaAuthSuccess()
+        }.catch { error in
+            self.handleOktaAuthFailure(error: error)
+        }
+        return false
+    }
+    
+    func signOutTappedForUITests() -> Bool {
+        guard let config = configForUITests else { return true }
+        OktaAuth.signOutOfOkta().start(withDictConfig: config, view: self).then {
+            self.handleOktaAuthLogoutSuccess()
+        }.catch { error in
+            self.handleOktaAuthFailure(error: error)
+        }
+        return false
+    }
+    
+    var configForUITests: [String: String]? {
```

Comment:
Probably better would be to have this stuff in tests and not in sample code. Just curios can we do the following:
- Add config property to view controller. If config is not nil then use it for browser signIn/signOut flows
- In test environment prepare test config
- Get reference to view controller and assign test config to config property. Then start required test flows

Reply:
Unfortunately, we don't have any control on application classes from `XCUIApplication` wrapper. We can only change launch arguments and environment variables. (https://developer.apple.com/documentation/xctest/xcuiapplication)
In current implementation we have config property in view controller which is nil by default end has value if we have corresponding env vars set by `XCUIApplication`

Reply:
Got it, thanks for analysis!